connection.go

```diff
@@ -426,6 +428,12 @@ func Connect(ctx context.Context, addr string, opts Opts) (conn *Connection, err
 	if err = conn.createConnection(ctx); err != nil {
 		return nil, err
 	}
+	conn.lockShards()
+	conn.schemaResolver = &noSchemaResolver{
+		isFeatureInSlice(iproto.IPROTO_FEATURE_SPACE_AND_INDEX_NAMES,
+			conn.serverProtocolInfo.Features),
+	}
+	conn.unlockShards()
```

Comment:
There are shouldn't be any concurrent requests yet.
```suggestion
	conn.schemaResolver = &noSchemaResolver{
		isFeatureInSlice(iproto.IPROTO_FEATURE_SPACE_AND_INDEX_NAMES,
			conn.serverProtocolInfo.Features),
	}
```

Reply:
Updated.

Reply:
Actually, there could be concurrent requests (in case of supported watchers). For example, https://github.com/tarantool/go-tarantool/actions/runs/6812197871/job/18524040071

Example:
```
WARNING: DATA RACE
Read at 0x00c0000dedb8 by goroutine 14:
  github.com/tarantool/go-tarantool/v2.(*Connection).putFuture()
      /home/tulchin/tarantool/go-tarantool/connection.go:1117 +0x6c4
  github.com/tarantool/go-tarantool/v2.(*Connection).send()
      /home/tulchin/tarantool/go-tarantool/connection.go:1095 +0x20b
  github.com/tarantool/go-tarantool/v2.(*Connection).Do()
      /home/tulchin/tarantool/go-tarantool/connection.go:1299 +0x8b
  github.com/tarantool/go-tarantool/v2.(*Connection).newWatcherImpl.func1()
      /home/tulchin/tarantool/go-tarantool/connection.go:1540 +0x484

Previous write at 0x00c0000dedb8 by main goroutine:
  github.com/tarantool/go-tarantool/v2.Connect()
      /home/tulchin/tarantool/go-tarantool/connection.go:437 +0xe7a
  github.com/tarantool/go-tarantool/v2/test_helpers.isReady()
      /home/tulchin/tarantool/go-tarantool/test_helpers/main.go:103 +0x157
  github.com/tarantool/go-tarantool/v2/test_helpers.StartTarantool()
      /home/tulchin/tarantool/go-tarantool/test_helpers/main.go:275 +0x950
  github.com/tarantool/go-tarantool/v2_test.runTestMain()
      /home/tulchin/tarantool/go-tarantool/tarantool_test.go:4093 +0x137
  github.com/tarantool/go-tarantool/v2_test.TestMain()
      /home/tulchin/tarantool/go-tarantool/tarantool_test.go:4105 +0x2e
  main.main()
      _testmain.go:453 +0x324

Goroutine 14 (running) created at:
  github.com/tarantool/go-tarantool/v2.(*Connection).newWatcherImpl()
      /home/tulchin/tarantool/go-tarantool/connection.go:1508 +0x295
  github.com/tarantool/go-tarantool/v2.(*Connection).dial()
      /home/tulchin/tarantool/go-tarantool/connection.go:581 +0x9d0
  github.com/tarantool/go-tarantool/v2.(*Connection).createConnection()
      /home/tulchin/tarantool/go-tarantool/connection.go:644 +0xa5
  github.com/tarantool/go-tarantool/v2.Connect()
      /home/tulchin/tarantool/go-tarantool/connection.go:428 +0xc95
  github.com/tarantool/go-tarantool/v2/test_helpers.isReady()
      /home/tulchin/tarantool/go-tarantool/test_helpers/main.go:103 +0x157
  github.com/tarantool/go-tarantool/v2/test_helpers.StartTarantool()
      /home/tulchin/tarantool/go-tarantool/test_helpers/main.go:275 +0x950
  github.com/tarantool/go-tarantool/v2_test.runTestMain()
      /home/tulchin/tarantool/go-tarantool/tarantool_test.go:4093 +0x137
  github.com/tarantool/go-tarantool/v2_test.TestMain()
      /home/tulchin/tarantool/go-tarantool/tarantool_test.go:4105 +0x2e
  main.main()
      _testmain.go:453 +0x324
```

Reply:
Or we could move the `conn.schemaResolver` init to the `dial` function (right after initing the `conn.serverProtocolInfo`).

Reply:
Please, move the setup here:

https://github.com/tarantool/go-tarantool/blob/e0e2b3e51b16ed2e7aec0e95c4100df641ac3e23/connection.go#L539